src/defineFetch/defineFetch.js

```diff
@@ -8,29 +8,34 @@ export function isFetchAction(action) {
   return action.type.startsWith(FETCH);
 }
 
-function memoize(actionCreatorFactory, make, conflict) {
+function memoize(displayName, actionCreatorFactory, make, conflict) {
   // TODO: may need a way to clear this memo
   const memo = {};
 
   function memoized(...keyArgs) {
-    const keyResult = make(...keyArgs);
+    const makeResult = make(...keyArgs);
 
-    if (typeof keyResult !== 'object') {
+    if (typeof makeResult !== 'object') {
       throw new Error('[defineFetch]: `make` must return an object');
     }
 
-    const { key, request } = keyResult;
+    const { request } = makeResult;
 
-    if (!Array.isArray(key)) {
-      throw new Error('[defineFetch] `key` must be an array in the object that `make` returns');
+    if (!keyArgs.every(key => typeof key === 'string' || typeof key === 'number')) {
+      const rogueKey = keyArgs.find(key => typeof key !== 'string' && typeof key !== 'number');
+      throw new Error(
+        `[defineFetch] make arguments must be either a string or a number. Found "${JSON.stringify(
+          rogueKey,
+        )}" for the fetch factory "${displayName}"`,
+      );
```

Comment:
I think `.find` would only return the first key found that meets the condition. Should we use filter instead and stringify the whole array found?

Reply:
this landed here: https://github.com/JustSift/ReSift/commit/e8e76c0f08d8f5bc9b38ea8a62af7f903bffb660